contrib/dockerd-rootless-setuptool.sh

```diff
@@ -0,0 +1,439 @@
+		DOCKERD_ROOTLESS_SH_FLAGS="${DOCKERD_ROOTLESS_SH_FLAGS} --iptables=false"
+	fi
+
+	# instruction: Debian and Arch require setting unprivileged_userns_clone
+	if [ -f /proc/sys/kernel/unprivileged_userns_clone ]; then
+		if [ "1" != "$(cat /proc/sys/kernel/unprivileged_userns_clone)" ]; then
+			instructions=$(
+				cat <<- EOI
+					${instructions}
+					# Set kernel.unprivileged_userns_clone
+					cat <<EOT > /etc/sysctl.d/50-rootless.conf
+					kernel.unprivileged_userns_clone = 1
+					EOT
+					sysctl --system
+				EOI
+			)
+		fi
+	fi
+
+	# instruction: RHEL/CentOS 7 requires setting max_user_namespaces
+	if [ -f /proc/sys/user/max_user_namespaces ]; then
+		if [ "0" = "$(cat /proc/sys/user/max_user_namespaces)" ]; then
+			instructions=$(
+				cat <<- EOI
+					${instructions}
+					# Set user.max_user_namespaces
+					cat <<EOT > /etc/sysctl.d/51-rootless.conf
+					user.max_user_namespaces = 28633
+					EOT
+					sysctl --system
+				EOI
+			)
+		fi
+	fi
+
+	# instructions: validate subuid/subgid files for current user
+	if ! grep -q "^$(id -un):\|^$(id -u):" /etc/subuid 2> /dev/null; then
+		instructions=$(
+			cat <<- EOI
+				${instructions}
+				# Add subuid entry for $(id -un)
+				echo "$(id -un):100000:65536" >> /etc/subuid
+			EOI
+		)
+	fi
+	if ! grep -q "^$(id -un):\|^$(id -u):" /etc/subgid 2> /dev/null; then
+		instructions=$(
+			cat <<- EOI
+				${instructions}
+				# Add subgid entry for $(id -un)
+				echo "$(id -un):100000:65536" >> /etc/subgid
+			EOI
+		)
+	fi
+
+	# fail with instructions if requirements are not satisfied.
+	if [ -n "$instructions" ]; then
+		ERROR "Missing system requirements. Run the following commands to"
+		ERROR "install the requirements and run this tool again."
+		if [ -n "$faced_iptables_error" ] && [ -z "$OPT_SKIP_IPTABLES" ]; then
+			ERROR "Alternatively iptables checks can be disabled with --skip-iptables ."
+		fi
+		echo
+		echo "########## BEGIN ##########"
+		echo "sudo sh -eux <<EOF"
+		echo "$instructions" | sed -e '/^$/d'
+		echo "EOF"
+		echo "########## END ##########"
+		echo
+		exit 1
+	fi
+	# TODO: support printing non-essential but recommended instructions:
+	# - sysctl: "net.ipv4.ping_group_range"
+	# - sysctl: "net.ipv4.ip_unprivileged_port_start"
+	# - external binary: slirp4netns
+	# - external binary: fuse-overlayfs
+}
+
+# CLI subcommand: "check"
+cmd_entrypoint_check() {
+	# requirements are already checked in init()
+	INFO "Requirements are satisfied"
+}
+
+# install (systemd)
+install_systemd() {
+	mkdir -p "${CFG_DIR}/systemd/user"
+	unit_file="${CFG_DIR}/systemd/user/${SYSTEMD_UNIT}"
+	if [ -f "${unit_file}" ]; then
+		WARNING "File already exists, skipping: ${unit_file}"
+	else
+		INFO "Creating ${unit_file}"
+		cat <<- EOT > "${unit_file}"
+			[Unit]
+			Description=Docker Application Container Engine (Rootless)
+			Documentation=https://docs.docker.com/engine/security/rootless/
+
+			[Service]
+			Environment=PATH=$BIN:/sbin:/usr/sbin:$PATH
```

Comment:
Oh shoot, just realizing this probably should've been `\$PATH`, sorry:

```suggestion
			Environment=PATH=$BIN:/sbin:/usr/sbin:\$PATH
```

Reply:
(I don't think this is critical though, which is why I've approved.)

Reply:
You mean the PATH needs to be propagated from the systemd --user, not from the current shell?

Reply:
Yeah, I think that would be the correct answer here (it also matches what we recommend below when the user isn't using systemd and has to run it manually themselves).

Reply:
`\$PATH` doesn't work as it sets PATH to `PATH=/usr/local/bin:/sbin:/usr/sbin:$PATH` without expanding `$PATH`

Reply:
Doh, ok, scratch this :see_no_evil: :innocent: